experiment_tool/dag.py

```diff
@@ -35,19 +51,48 @@ def dfs(x):
                 if y not in visited:
                     dfs(y)
             res.append(x)
-
         for v in self._graph:
             if v not in visited:
                 dfs(v)
         res.reverse()
+        log.debug('finish topological sort')
         return res
 
     def get_subgraph(self, metrics, modified):
         """
         Return subgraph that contains all vertices, functions in which should be recalculated
         in order to obtain desired metrics.
         """
-        return self
+        vertices = set()
+        edges = []
+        visited = set()
+
+        def dfs(x):
+            visited.add(x)
+            recalculate = False
+            if x in metrics:
+                recalculate = True
+            for y in self._graph[x]:
+                if y not in visited:
+                    if dfs(y):
+                        recalculate = True
+                elif y in vertices:
+                    recalculate = True
+            if recalculate:
+                vertices.add(x)
+                return True
+            return False
```

Comment:
```suggestion
            return recalculate
```
Можно упростить 83-84 строчки

Reply:
done